eventhub_endpoint.bal

```diff
@@ -201,13 +190,19 @@ public client class Client {
     # Retrieves all metadata associated with all Event Hubs within a specified Service Bus namespace
     #
     # + return - Return list of event hubs or error
-    public remote function listEventHubs() returns @tainted xml|error {
-        http:Request req = self.getAuthorizedRequest();
+    remote function listEventHubs() returns @tainted xml|error {
+        http:Request req = getAuthorizedRequest(self.config);
         var response = self.clientEndpoint->get("/$Resources/EventHubs", req);
         if (response is http:Response) {
-            var xmlPayload = response.getXmlPayload();
-            if (xmlPayload is xml) {
-                return xmlPayload;
+            var textPayload = response.getTextPayload();
+            if (textPayload is string) {
+                string cleanedStringXMLObject = stringutils:replaceAll(textPayload, "xml:base", "xml");
+                xml|error xmlPayload = 'xml:fromString(cleanedStringXMLObject);
+                if (xmlPayload is xml) {
+                    return xmlPayload;
+                } else {
+                    return Error(xmlPayload.message()); 
+                }         
             }
             return Error("invalid response from EventHub API. status code: " + response.statusCode.toString()
```

Comment:
Error messages seem repetitive. We can use constants.

Reply:
solved by using a common error function to get the response payload and response status code